### ISSUE TITLE
Multiple namespaces can be problematic

### ISSUE THREAD

**Issue by bermud** [opengeospatial/ets-wfs10#2]
The test uses 2 namespaces cdf and cgf which can be problematic. Change to have only one.

These issue involves these tests:

 Test wfs:test1.0.0-basic-describefeaturetype-get-4 (s0001/d1e712_1): Failed
 Test wfs:test1.0.0-basic-describefeaturetype-get-7 (s0001/d1e737_1): Failed
 Test wfs:test1.0.0-basic-describefeaturetype-post-4 (s0001/d1e786_1): Failed
 Test wfs:test1.0.0-basic-describefeaturetype-post-7 (s0001/d1e810_1): Failed

**Comment by segoerke** [comment]
I see, this can be a problem if implementations aren't able to handle such data. The question is, if the capability of handling multiple namespaces is a recommendation or a requirement.

**Comment by bermud** [comment]
@dstenger, please investigate if this is a real requirement, if not please simplify.

**Comment by dstenger** [comment]
For futher analysis we need a service URL or a DescribeFeatureType response of a service with multiple namespaces which is failing the tests mentioned above.

In general, the tests should be capable to handle services with multiple namespaces.

**Comment by bermud** [comment]
I will check with Satish for him to provide a service to test with

**Comment by bermud** [comment]
In process of getting a public server to test.

**Comment by bermud** [comment]
It is being hard to get an online server, but the logs are enclosed. Please review.
[s0001.zip](https://github.com/opengeospatial/ets-wfs10/files/106904/s0001.zip)

**Comment by dstenger** [comment]
We analysed the log and following assertion causes the test failure:
```
[FAILED] XPath expression is 'true': not(boolean(/xs:schema) and citef:validate-gml-schema('http://arcgis-agreleaseusweorwinsql1-1849550334.us-west-2.elb.amazonaws.com/arcgis/services/wfs100/MapServer/WFSServer?',/xs:schema, 'cdf:Deletes', 'cgf:Points'))
```

The assertion fails as the Points element is bound to the cdf namespace instead of cgf.
Can you please review the server configuration, why Points is bound to cdf insead of cgf?

**Comment by bermud** [comment]
Please look at the standard section 8.3.1 to check if the test data follows the standard correctly.

**Comment by dstenger** [comment]
The cdf and cgf namespaces are hard coded in the test:
```
<xsl:variable name="expression">not(boolean(/xs:schema) and citef:validate-gml-schema('<xsl:value-of select="$VAR_WFS_DESCRIBE_FEATURE_TYPE_HTTP_GET_URL"/>',/xs:schema, 'cdf:Deletes', 'cgf:Points'))</xsl:variable>
```

So, this test fails as the service configuration does not match the prerequisites.

**Comment by dstenger** [comment]
In the test data these elements are bound to two namespaces. Section 8.3.1 explains how to integrate multiple namespaces.Thus, the test data follow the standard correctly.

**Comment by bermud** [comment]
@dstenger. Ok I will close this issue.

**Comment by dstenger** [comment]
The specification say following regarding namespaces:

```
6.6 Namespaces

Namespaces (17) are used to discriminate XML vocabularies from one another. For the
WFS there are three normative namespace definitions, namely:

• (http://www.opengis.net/wfs) - for the WFS interface vocabulary
• (http://www.opengis.net/gml) - for the GML vocabulary
• (http://www.opengis.net/ogc) - for the OGC Filter vocabulary

A given WFS implementation will make use of one or more GML Application Schemas
and these schemas will use, in turn, one or more application namespaces (e.g.
http://www.someserver.com/myns). While many of the examples in this document use a
single namespace, multiple namespaces can be used, as shown section 11.2.6.
```

```
8.3.1 Supporting multiple namespaces

An XML Schema[6] document can only describe elements that belong to a single
namespace. This means that a Web Feature Service cannot describe features from
multiple namespaces in a single XML Schema document. To overcome this limitation, a
WFS may generate an XML Schema document that is a “wrapper” schema that imports
the schemas of the features from the various namespaces in the request. For example,
consider the following request:

see specification, page 19

In this example, the WFS is using a DescribeFeatureType request to obtain the schemas
of the features in the various namespaces. This is simply an example, other methods of
obtaining the schemas may be implemented (for example: referencing static schema
documents).
```

In my understanding, this states that multiple namespaces should be supported (especially if the last sentence of chapter 6.6 is considered).
To clarify this issue to 100%, it should be considered to contact the corresponding WG.

**Comment by bermud** [comment]
I think we should relax the test. The specification says 1 or more. Maybe we can test for the other namespace but provide a warning instead of a fail.

**Comment by bermud** [comment]
Maybe move all these test to a   'multiple namespaces'  conformance class, that are not part of the 'basic'. Need to check that these tests do not contain other test requirements.

**Comment by dstenger** [comment]
Two tests verify whether  multiple namespaces are supported. Those tests can be moved to a 'multiple namespaces' conformance class.

The other two tests are checking more general requirements. So these tests have to be made more lax.

**Comment by dstenger** [comment]
As test data and tests must be updated, a new issue was created for the implementation: https://github.com/opengeospatial/ets-wfs10/issues/11

This issue can be closed as soon as the other one is solved.

**Comment by dstenger** [comment]
Tests were moved in a new conformance class and two test were made more lax.

Fix is provided via pull request #13.
